wasm/src/types/integer.rs

```diff
@@ -0,0 +1,255 @@
+                Self(self.0.add_wrapped(&other.0))
+            }
+
+            /// Wrapped subtraction.
+            #[wasm_bindgen(js_name = "subWrapped")]
+            pub fn sub_wrapped(&self, other: &$name) -> $name {
+                Self(self.0.sub_wrapped(&other.0))
+            }
+
+            /// Wrapped multiplication.
+            #[wasm_bindgen(js_name = "mulWrapped")]
+            pub fn mul_wrapped(&self, other: &$name) -> $name {
+                Self(self.0.mul_wrapped(&other.0))
+            }
+
+            /// Wrapped division.
+            #[wasm_bindgen(js_name = "divWrapped")]
+            pub fn div_wrapped(&self, other: &$name) -> $name {
+                Self(self.0.div_wrapped(&other.0))
+            }
+
+            /// Power to a u8 exponent.
+            #[wasm_bindgen(js_name = "powU8")]
+            pub fn pow_u8(&self, exponent: &U8) -> $name {
+                Self(self.0.pow(&exponent.0))
+            }
+
+            /// Power to a u16 exponent.
+            #[wasm_bindgen(js_name = "powU16")]
+            pub fn pow_u16(&self, exponent: &U16) -> $name {
+                Self(self.0.pow(&exponent.0))
+            }
+
+            /// Power to a u32 exponent.
+            #[wasm_bindgen(js_name = "powU32")]
+            pub fn pow_u32(&self, exponent: &U32) -> $name {
+                Self(self.0.pow(&exponent.0))
+            }
+
+            /// Negates the integer (e.g., 5 → -5).
+            #[wasm_bindgen(js_name = "neg")]
+            pub fn neg(&self) -> $name {
+                Self(-self.0)
+            }
+
+            /// Checks equality with another integer.
+            #[wasm_bindgen(js_name = "equals")]
+            pub fn equals(&self, other: &$name) -> bool {
+                self.0 == other.0
+            }
+
+            /// Remainder.
+            #[wasm_bindgen(js_name = "rem")]
+            pub fn rem(&self, other: &$name) -> $name {
+                Self(self.0.rem(&other.0))
+            }
+
+            /// Wrapped remainder.
+            #[wasm_bindgen(js_name = "remWrapped")]
+            pub fn rem_wrapped(&self, other: &$name) -> $name {
+                Self(self.0.rem_wrapped(&other.0))
+            }
+
+            /// Convert to Scalar.
+            #[wasm_bindgen(js_name = "toScalar")]
+            pub fn to_scalar(&self) -> crate::Scalar {
+                self.0.to_scalar().into()
+            }
+
+            /// Convert to plaintext.
+            #[wasm_bindgen(js_name = "toPlaintext")]
+            pub fn to_plaintext(&self) -> Plaintext {
+                Plaintext::from(PlaintextNative::Literal(LiteralNative::$name(self.0), OnceCell::new()))
+            }
+
+            /// Convert from Field.
+            #[wasm_bindgen(js_name = "fromField")]
+            pub fn from_field(field: &crate::Field) -> Result<$name, String> {
+                <$native>::from_field(field.as_native()).map(Self).map_err(|e| e.to_string())
```

Comment:
Use the inherent From trait.

```suggestion
                <$native>::from_field(FieldNative::from(field)).map(Self).map_err(|e| e.to_string())
```